webxrlayers-1.bs

```diff
@@ -0,0 +1,730 @@
+the rectangular dimensions of the {{XRLayer}}
+
+The <dfn dict-member for="XRLayerInit">depth</dfn> attribute defines if the {{XRLayer}} view will have a depth buffer.
+
+The <dfn dict-member for="XRLayerInit">stencil</dfn> attribute defines if the {{XRLayer}} view will have a stencil buffer.
+
+The <dfn dict-member for="XRLayerInit">alpha</dfn> attribute defines if the {{XRLayer}} view will have an alpha channel.
+
+XRWebGLLayerFactory {#xrwebgllayerfactorytype}
+-------------------
+The {{XRWebGLLayerFactory}} object is used to create layers that have a GPU backend.
+
+<pre class="idl">
+interface XRWebGLLayerFactory {
+  constructor(XRSession session, XRWebGLRenderingContext context);
+
+  readonly attribute double nativeProjectionScaleFactor;
+
+  Promise&lt;XRProjectionLayer&gt; requestProjectionLayer(GLenum textureTarget, 
+                                                   optional XRProjectionLayerInit init);
+  Promise&lt;XRQuadLayer&gt; requestQuadLayer(GLenum textureTarget, 
+                                       XRLayerInit init);
+  Promise&lt;XRCylinderLayer&gt; requestCylinderLayer(GLenum textureTarget, 
+                                               XRLayerInit init);
+  Promise&lt;XREquirectLayer&gt; requestEquirectLayer(GLenum textureTarget, 
+                                               XRLayerInit init);
+  Promise&lt;XRCubeLayer&gt; requestCubeLayer(XRLayerInit init);
+
+  XRWebGLSubImage? getSubImage(XRLayer layer); // for mono layers
+  XRWebGLSubImage? getViewSubImage(XRLayer layer, XRView view); // for stereo layers
+};
+</pre>
+
+Each {{XRWebGLLayerFactory}} has a <dfn for="XRWebGLLayerFactory">context</dfn> object which is an instance 
+of either a {{WebGLRenderingContext}} or a {{WebGL2RenderingContext}}.
+
+Each {{XRWebGLLayerFactory}} has an associated <dfn for="XRWebGLLayerFactory">session</dfn>, which is the 
+{{XRSession}} it was created with.
+
+Each {{XRLayer}} created through {{XRWebGLLayerFactory}} has an internal <dfn for="XRLayer">colorTextures</dfn> which is an |array| of {{WebGLTexture}} for color textures and 
+an internal <dfn for="XRLayer">depthStencilTextures</dfn> whcih is an |array| of {{WebGLTexture}}</dfn>for depth/stencil textures.
+
+<div class="algorithm" data-algorithm="construct-webgl-layer">
+
+The <dfn constructor for="XRWebGLLayerFactory">XRWebGLLayerFactory(|session|, |context|)</dfn> constructor 
+MUST perform the following steps when invoked:
+
+  1. Let |factory| be a new {{XRWebGLLayerFactory}}
+  1. If |session|'s [=ended=] value is <code>true</code>, throw an {{InvalidStateError}} and abort these steps.
+  1. If |context| is lost, throw an {{InvalidStateError}} and abort these steps.
+  1. If |session| is not an [=immersive session=] or |context|'s [=XR compatible=] boolean is <code>false</code>, throw an {{InvalidStateError}} and abort these steps.
```

Comment:
Just for consistency it seems like this line should be two separate steps, because the checks aren't really related to each other.

Reply:
done